src/marqo/tensor_search/add_docs.py

```diff
@@ -61,59 +87,202 @@ def threaded_download_and_preprocess_images(allocated_docs: List[dict], image_re
+                            try:
+                                content_repo[doc[field]] = preprocessors['image'](content_repo[doc[field]]).to(device)
+                                print(f"image_repo[doc[field]]: {content_repo[doc[field]]}")
+                            except OSError as e:
+                                if "image file is truncated" in str(e):
+                                    content_repo[doc[field]] = e
+                                    metric_obj.increment_counter(f"{doc.get(field, '')}.OSError")
+                                    continue
+                                else:
+                                    raise e
+
+                    if modality in [Modality.VIDEO, Modality.AUDIO]:
+                        chunks = download_and_chunk_media(doc[field], download_headers, memory_pool, modality)
+                        content_repo[doc[field]] = chunks
+
                 # For multimodal tensor combination
                 elif isinstance(doc[field], dict):
                     for sub_field in list(doc[field].values()):
                         if isinstance(sub_field, str) and clip_utils._is_image(sub_field):
-                            if sub_field in image_repo:
+                            if sub_field in content_repo:
                                 continue
                             try:
-                                image_repo[sub_field] = clip_utils.load_image_from_path(
+                                content_repo[sub_field] = clip_utils.load_image_from_path(
                                     sub_field,
                                     image_download_headers,
                                     timeout=TIMEOUT_SECONDS,
                                     metrics_obj=metric_obj
                                 )
                             except PIL.UnidentifiedImageError as e:
-                                image_repo[sub_field] = e
+                                content_repo[sub_field] = e
                                 metric_obj.increment_counter(f"{doc.get(field, '')}.UnidentifiedImageError")
                                 continue
 
 
+def download_and_chunk_media(url: str, headers: dict, memory_pool: MemoryPool, modality: Modality) -> List[np.ndarray]:
+    chunks = []
+    buffer = BytesIO()
+
+    def write_function(data):
+        buffer.write(data)
+        if buffer.tell() >= memory_pool.chunk_size:
+            chunk = memory_pool.get_chunk()
+            buffer.seek(0)
+            chunk[:len(buffer.getvalue())] = buffer.getvalue()
+            chunks.append(process_chunk(chunk, modality))
+            memory_pool.return_chunk(chunk)
+            buffer.seek(0)
+            buffer.truncate()
+
+    c = pycurl.Curl()
+    c.setopt(c.URL, url)
+    c.setopt(c.WRITEFUNCTION, write_function)
+    for key, value in headers.items():
+        c.setopt(c.HTTPHEADER, [f"{key}: {value}"])
```

Comment:
have you tested if headers will be added? This is what we did before: `c.setopt(pycurl.HTTPHEADER, [f"{k}: {v}" for k, v in headers.items()])`